instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py

```diff
@@ -204,10 +211,61 @@ def collect_request_attributes(environ):
         flavor = flavor[len(_HTTP_VERSION_PREFIX) :]
     if flavor:
         result[SpanAttributes.HTTP_FLAVOR] = flavor
+    capture_custom_request_headers(
+        environ,
+        result,
+    )
 
     return result
 
 
+def capture_custom_request_headers(environ, attributes):
+    """Collects custom HTTP request headers configured by the user from the PEP3333-conforming
+    WSGI environ and adds it to attributes dictionary to be used as span creation attributes as described
+    in the specification https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers"""
+
+    custom_request_headers = get_custom_headers(
+        OTEL_PYTHON_CAPTURE_REQUEST_HEADERS
+    )
+    for header_name in custom_request_headers:
+        wsgi_env_var = header_name.upper().replace("-", "_")
+        header_values = environ.get(f"HTTP_{wsgi_env_var}")
+
+        if header_values:
+            key = normalise_request_header_name(header_name)
+            attributes[key] = [
+                header_value.strip()
+                for header_value in header_values.split(",")
```

Comment:
How does this work when there is , in header value?

Reply:
Yes, in this case we won't get the correct value of the header. I chose ',' as delimiter because it is mentioned in the http spec that multiple header values should be concatenated with comma. ([link](https://stackoverflow.com/questions/3096888/standard-for-adding-multiple-values-of-a-single-http-header-to-a-request-or-resp))
But, I agree client can pass ',' in single header value. 
In case we don't want to split the header_values into multiple values we can just pass the whole string as a single element in the list. ([otel-spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers) allows that).
Please share your thoughts.

Reply:
> either multiple header values as an array of strings or a single-item array containing a possibly comma-concatenated string, depending on the way the HTTP library provides access to headers.

Since the wsgi environ object gives the single value only, I think it's better we use the whole header value as attribute.

Reply:
Change done.